### PR TITLE
chore: setup canonical meta

### PR DESCRIPTION
**Pull request by samouss**
**Summary**

This PR setup the meta `canonical` for all the pages of the website. If no meta `canonical` is found for the page we fallback to an `noindex, nofollow` meta. Here is the list of changes:

- allow `canonical` meta or fallback on each pages
- setup `canonical` meta tags for top level pages & guides
- setup `canonical` meta tags for the API Ref

### CODE REVIEW

**Comment by samouss** [comment]
@Haroenv Yes indeed, but like you said it's for Google I think it's fine.

File: docgen/src/widgets.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ withHeadings: true
 navWeight: 100
 editable: true
 githubSource: docgen/src/widgets.md
+canonical: https://www.algolia.com/doc/guides/building-search-ui/widgets/showcase/js/
```
Comment (tkrugg) [inline docgen/src/widgets.md]: there is also this one that can be used here: https://www.algolia.com/doc/api-reference/widgets/js/
No sure which one we should prefer.

Comment (Haroenv) [reply docgen/src/widgets.md]: the showcase is a nicer link

Comment (samouss) [reply docgen/src/widgets.md]: I've asked Marie about this one, definitely the showcase!

**Pull request merged**